modules/nf-core/gmmdemux/main.nf

```diff
@@ -0,0 +1,68 @@
+
+process GMMDEMUX {
+    tag "$meta.id"
+    label 'process_low'
+
+    conda "${moduleDir}/environment.yml"
+    container "${ workflow.containerEngine == 'singularity' && !task.ext.singularity_pull_docker_container ?
+        'https://depot.galaxyproject.org/singularity/gmm-demux:0.2.2.3--pyh7cba7a3_0':
+        'biocontainers/gmm-demux:0.2.2.3--pyh7cba7a3_0' }"
+
+    input:
+    tuple val(meta), path(hto_matrix),val(hto_names)
+    val type_report
+    val summary_report
+    val skip
+
+    output:
+    tuple val(meta), path("${meta.id}/barcodes.tsv.gz"                          ), emit: barcodes
+    tuple val(meta), path("${meta.id}/*.mtx.gz"                                 ), emit: matrix
+    tuple val(meta), path("${meta.id}/features.tsv.gz"                          ), emit: features
+    tuple val(meta), path("${meta.id}/classification_report_${meta.id}"         ), emit: classification_report
+    tuple val(meta), path("summary_report_${meta.id}.txt"                       ), emit: summary_report, optional: true
```

Comment:
The problem is summary_report, this is an optional output, that is produced only when it gets an existing file to store the data, defined as:` Specify the file to store summary report. Require a file argument. `so the problem is that the directory where the other outputs are stored is created after this parameter is analysed. Then, when I tried to give it the same path as the others, something like: `"${meta.id}/"summary_report_${meta.id}.txt"` I get an error, because the folder doesn't exist.

Reply:
Okay, I think I understand. I guess if you wanted everything in the same folder, you could move the file after running the tool.

Reply:
ok, if you agree I also added some code to check if the tool wrote something in the summary report file before moving it to the same folder of the results, as I mentioned above, the tool not always writes the report. When the output says: 
`GMM cannot find a viable solution that satisfies the droplet formation model.`
The summary remains empty.